common/djangoapps/student/tests/tests.py

```diff
@@ -930,3 +932,190 @@ def test_roundtrip_with_unicode_course_id(self):
+                'status': course[1],
+                'course_codes': [
+                    {
+                        'organization': {'display_name': 'Test Organization 1', 'key': 'edX'},
+                        'marketing_slug': 'manual-test-1',
+                        'display_name': 'Manual Test',
+                        'key': 'TEST_A',
+                        'run_modes': [
+                            {'sku': '', 'mode_slug': 'ABC_1', 'course_key': 'edX/DemoX_1/Run_1'},
+                            {'sku': '', 'mode_slug': 'ABC_2', 'course_key': 'edX/DemoX_2/Run_2'},
+                            {'sku': '', 'mode_slug': 'ABC_3', 'course_key': 'edX/DemoX_2/Run_3'},
+                        ]
+                    }
+                ],
+                'subtitle': 'Dummy program 1 for testing',
+                'name': 'First Program'
+            }
+
+        return programs
+
+    @ddt.data(
+        ('active', [{'sku': ''}, {'sku': ''}, {'sku': ''}, {'sku': ''}], 4, 'slug1'),
+        ('active', [{'sku': ''}, {'sku': ''}, {'sku': ''}], 3, 'slug2'),
+        ('active', [], 0, ''),
+        ('inactive', [{'sku': ''}, {'sku': ''}, {'sku': ''}, {'sku': ''}], 4, 'slug3'),
+    )
+    @ddt.unpack
+    def test_get_xseries_programs_method(self, status, run_modes, run_modes_count, slug):
+        with patch('student.views.dummy_method_for_test_cases') as mock_data:
+            mock_data.return_value = {
+                unicode('course-v1:ManTestX+ManTest2+2014'): {
+                    'category': 'xseries',
+                    'status': status,
+                    'course_codes': [
+                        {
+                            'organization': {'display_name': 'Test Organization 1', 'key': 'edX'},
+                            'marketing_slug': slug,
+                            'display_name': 'Manual Test',
+                            'key': 'TEST_A',
+                            'run_modes': run_modes
+                        }
+                    ],
+                    'subtitle': 'Dummy program 1 for testing',
+                    'name': 'First Program'
+                }
+            }
+            parse_data = _get_xseries_programs(
+                self.user, [unicode('course-v1:ManTestX+ManTest2+2014'), 'edx/demox/DemoCourse']
+            )
+
+            if status == 'inactive':
+                self.assertDictEqual({}, parse_data)
+            else:
+                self.assertDictEqual(
+                    {
+                        'course-v1:ManTestX+ManTest2+2014': {
+                            'run_modes_count': run_modes_count,
+                            'marketing_slug': slug,
+                            'display_name': 'Manual Test',
+                            'category': 'xseries'
+                        }
+                    },
+                    parse_data
+                )
+
+    @unittest.skipUnless(settings.ROOT_URLCONF == 'lms.urls', 'Test only valid in lms')
```

Comment:
Since you are writing tests for LMS dashboard so it make sense to move this settings to class `DashboardTestXSeriesPrograms` level